### PR TITLE
Make `--incompatible_remote_build_event_upload_respect_no_cache` a no-op

### PR DESCRIPTION
**Pull request by brentleyjones**
Also removed unneeded tests.

RELNOTES: `--incompatible_remote_build_event_upload_respect_no_cache` is now a no-op.

### CODE REVIEW

**Review by coeuvre — APPROVED** [review]

**Comment by brentleyjones** [comment]
Any context on the delay of this import @kshyanashree?

**Comment by kshyanashree** [comment]
> Any context on the delay of this import @kshyanashree?

Hi @brentleyjones! There were conflicts where I removed few unneeded tests. Still presubmits are failing, Could you please look into it?

**Comment by kshyanashree** [comment]
> I'll rebase soon.

Thank you!

File: src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java [hunk 1]
```diff
@@ -757,8 +757,7 @@ public void afterAnalysis(
   // Separating the conditions for readability.
   private boolean shouldParseNoCacheOutputs() {
     if (remoteOptions == null
-        || remoteOptions.remoteBuildEventUploadMode != RemoteBuildEventUploadMode.ALL
-        || !remoteOptions.incompatibleRemoteBuildEventUploadRespectNoCache) {
```
Comment (brentleyjones) [inline src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java:761]: @coeuvre The way this was written, using one flag or the other, meant that this always evaluated to `true` (so always returned `false`). The tests are breaking with this change, and to get the hold behavior, I would instead change this to `remoteOptions == null || true`, or just have the function always return `false`.

Not sure what to do for this.

Comment (coeuvre) [reply src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java:761]: always return `false` SGTM.

**Review by coeuvre — APPROVED** [review]

**Comment by brentleyjones** [comment]
@kshyanashree all green now

**Comment by kshyanashree** [comment]
> @kshyanashree all green now

Thanks @brentleyjones!

**Pull request closed without merging**